### PR TITLE
Unification of JSON RPC codebase

### PR DESCRIPTION
**Pull request by dkulieshov**
### What does this PR do?
We had two different implementations of JSON RPC subsystem because of different Java version used on client and server side. After we updated java version on client we no more need different implementations. This PR is intended to make code base as unified as possible for both sides. More accurately current difference is mostly within JSON parsing libraries (which are different for client and server sides).

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/4616

#### Changelog
Client side (IDE) and server side (workspace agent, workspace master) JSON RPC services codebase is unified

#### Release Notes
Client side (IDE) and server side (workspace agent, workspace master) JSON RPC services codebase is unified

### CODE REVIEW

**Review by vparfonov — APPROVED** [review]

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcMarshaller.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     *         response
+     * @return string representation
+     */
+    String marshall(JsonRpcResponse response);
+
+    /**
+     * Serializes JSON RPC request object into a string
+     *
+     * @param request
+     *         response
```
Comment (RomanNikitenko) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcMarshaller.java]: please fix typo

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/RequestHandlerManager.java [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+ * <ul>
+ *     <li>{@link NoneToNoneHandler} - to receive a notification w/o parameters</li>
+ *     <li>{@link NoneToOneHandler} - to receive a request w/o parameters and a single result</li>
+ *     <li>{@link NoneToManyHandler} - to receive a request w/o parameters and multiple results</li>
+ *     <li>{@link OneToNoneHandler} - to receive a notification with a single parameter</li>
+ *     <li>{@link OneToOneHandler} - to receive a request with a single parameter and a single result</li>
+ *     <li>{@link OneToManyHandler}- to receive a request with a single parameter and multiple results </li>
+ *     <li>{@link ManyToNoneHandler} - to receive a notification with multiple parameters</li>
+ *     <li>{@link ManyToOneHandler} - to receive request with multiple parameters and a single result</li>
+ *     <li>{@link ManyToManyHandler} - to receive request with multiple parameters and mu</li>
```
Comment (RomanNikitenko) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/RequestHandlerManager.java]: the sentence is not finished, I suppose: "...and multiple results"

**Review by RomanNikitenko — APPROVED** [review]

**Review by svor — APPROVED** [review]

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcQualifier.java [hunk 3]
```diff
@@ -8,12 +8,12 @@
  * Contributors:
  *   Codenvy, S.A. - initial API and implementation
  *******************************************************************************/
-package org.eclipse.che.api.core.jsonrpc;
+package org.eclipse.che.api.core.jsonrpc.commons;
```
Comment (skabashnyuk) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcQualifier.java:12]: javadoc

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/RequestHandlerManager.java [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+ *     <li>{@link NoneToManyHandler} - to receive a request w/o parameters and multiple results</li>
+ *     <li>{@link OneToNoneHandler} - to receive a notification with a single parameter</li>
+ *     <li>{@link OneToOneHandler} - to receive a request with a single parameter and a single result</li>
+ *     <li>{@link OneToManyHandler}- to receive a request with a single parameter and multiple results </li>
+ *     <li>{@link ManyToNoneHandler} - to receive a notification with multiple parameters</li>
+ *     <li>{@link ManyToOneHandler} - to receive request with multiple parameters and a single result</li>
+ *     <li>{@link ManyToManyHandler} - to receive request with multiple parameters and multiple results</li>
+ * </ul>
+ */
+@Singleton
```
Comment (skabashnyuk) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/RequestHandlerManager.java:44]: think about if this class is thread safe

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**